probe-rs-tools/src/bin/probe-rs/util/rtt/client.rs

```diff
@@ -182,11 +185,17 @@ impl RttClient {
     }
 
     pub(crate) fn clear_control_block(&mut self, core: &mut Core) -> Result<(), Error> {
-        self.try_attach(core)?;
+        let backup_address = match self.try_attach_with_address(core) {
+            Ok(_) => 0,
+            Err((e, None)) => return Err(e),
+            Err((_, Some(addr))) => addr,
+        };
 
         let Some(target) = self.target.as_mut() else {
-            // If we can't attach, we don't have a valid
-            // control block and don't have to do anything.
+            // If we can't attach, but we do have the address of the control block
+            let zeros = vec![0; Rtt::control_block_size(core)];
+            core.write(backup_address, &zeros)?;
```

Comment:
I don't know, clearing a random bit of memory that is currently not a control block doesn't seem polite toward bootloaders.